mesonbuild/modules/pkgconfig.py

```diff
@@ -342,12 +342,21 @@ def _generate_pkgconfig_file(self, state, deps, subdirs, name, description,
         libdir = PurePath(coredata.get_option(mesonlib.OptionKey('libdir')))
         incdir = PurePath(coredata.get_option(mesonlib.OptionKey('includedir')))
         fname = os.path.join(outdir, pcfile)
+
+        # check if any variables reference libdir, to make sure we include it
+        # when its needed
+        need_libdir = False
+
+        for k, v in variables + unescaped_variables:
+            if '${libdir}' in v:
```

Comment:
As mentioned in a review comment to my PR, this will match if the value is `${prefix}/${libdir}`.

My response to that was I *intentionally* only allow it at the beginning, because it's not allowed to appear in the middle -- that would produce broken paths, and thus is a bug in the user's `meson.build` which Meson should not cater to.